annotated_types/__init__.py

```diff
@@ -237,7 +291,18 @@ class MultipleOf(BaseMetadata):
     and libraries to carefully document which they implement.
     """
 
-    multiple_of: Union[SupportsDiv, SupportsMod]
+    multiple_of: T
+
+    @overload
+    def __supports_type__(self, obj: SupportsDiv[T]) -> bool:
+        ...
+
+    @overload
+    def __supports_type__(self, obj: SupportsMod[T]) -> bool:
+        ...
+
+    def __supports_type__(self, obj: SupportsMod[T] | SupportsDiv[T]) -> bool:
+        raise NotImplementedError
```

Comment:
I'm choosing to not deal with this case and leaving it up to implementers. I think the overloads are correct for static type checkers?

Reply:
I don't think you even need the overloads.

Reply:
You're right I can just make it a union. I think I had originally implemented one of the two options.